test/test_genty.py

```diff
@@ -96,6 +96,25 @@ def test_decorated(self, sole_arg):
             )(),
         )
 
+    def test_genty_dataprovider_doesnt_need_any_datasets(self):
+        @genty
+        class SomeClass(object):
+            def my_param_factory(self):
+                return 101
+
+            @genty_dataprovider(my_param_factory)
+            def test_decorated(self, sole_arg):
+                return sole_arg
+
+        instance = SomeClass()
+        self.assertEqual(
+            101,
+            getattr(
+                instance,
+                'test_decorated_{0}'.format('my_param_factory'),
```

Comment:
a) All of the tests do this sort of composition of the 'name' of the fabricated test method. So this does the same.

b) And while instance.constant would work for this example, it doesn't when the method name include characters that would cause interpreter to barf. Example:
    instance.test_method(4, 7)() 
See line 95 for an example.